es6-shim.js

```diff
@@ -2097,6 +2097,88 @@
     // Shim incomplete iterator implementations.
     addIterator(Object.getPrototypeOf((new globals.Map()).keys()));
     addIterator(Object.getPrototypeOf((new globals.Set()).keys()));
+
+    // Reflect
+    if (!globals.Reflect) {
+      globals.Reflect = {
+
+        // Syntax in a functional form.
+        get: function (target, key, receiver) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          return target[key];
+        },
+
+        set: function (target, key, value, receiver) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          target[key] = value;
+        },
+
+        has: function (target, key) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          return key in target;
+        },
+
+        deleteProperty: function (target, key) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          delete target[key];
+        },
+
+        enumerate: function (target) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          var keys = [];
+
+          for (var key in target) {
+            keys.push(key);
+          }
+
+          return new ArrayIterator(keys, 'value');
```

Comment:
I think their implementation is wrong - https://people.mozilla.org/~jorendorff/es6-draft.html#sec-ordinary-object-internal-methods-and-internal-slots-enumerate clearly says if a property is deleted before it's iterated over, it's ignored, and that "if new properties are added to the object being enumerated during enumeration, the newly added properties are not guaranteed to be processed in the active enumeration." - so deleted properties must be ignored, but newly added ones don't necessarily.

This suggests that we might need, on each `next` call, to iterate via `for..in` or `Object.keys` until the "last seen key" is seen, and then return the key after that - ie, each `next` call returns something like `var keys = Object.keys(obj); var i = keys.indexOf(lastKey); lastKey = i; return keys[i + 1];`